### ISSUE TITLE
All Variable Types to be Settable/Saveable

### ISSUE THREAD

**Issue by Jacki3** [LoGaCulture/LUTE#64]
**Is your feature request related to a problem? Please describe.**
Currently we only allow for ints, floats and strings (with some custom variables) to be set and saved (persistent) whereas we should ensure that all variables allow for this.

**Describe the solution you'd like**
Ensure all functionality of all variables is possible and robust then have this auto setup for new variables when they are created.

**Additional context**
Ideally we do this automatically upon new variables being constructed but currently no implementation! It is a simple process to update them but takes time to go through each and test.

**Comment by Jacki3** [comment]
Also ensure you see other implementations of saving/loading variables - are we missing key orders for variables?
